### DISCUSSION TITLE
Składowanie logów w CSV z możliwością sporadycznego przeszukania

### DISCUSSION THREAD

**Discussion by dlxpbva** [AzureCommunityPL/group#56] [Pomoc]
Czołem grupo,

Szukam rozwiązania do taniego składowania logów SMTP, dane są w formie plików CSV, razem jest tego ok. 80 GB.

Zależy mi na możliwości przeszukania tych plików pod kątem danego słowa kluczowego bądź jego fragmentu (wyszukiwanie z wildcard). Wrzuciłem sobie testowo jeden plik do table storage i tam niestety wildcard nie działa. Dostęp do danych odbywał by się srednio raz na miesiąc, nie częściej. 

Macie jakieś sugestie?

**Comment by ptrstpp950** [comment]
Użyłbym storage tak jak chciałeś i wykorzystałbym możliwość podpinania go jako "dysk" (patrz: https://docs.microsoft.com/en-us/azure/storage/blobs/network-file-system-protocol-support-how-to?tabs=linux)

Do sprawdzenia czy taniej Ci wyjdzie przeszukiwanie przez VM stojącą w Azure, czy przez lokalną maszynę. Dzieki VM w Azure zaoszczędzisz na transferze danych, ale zapłacisz za VM'kę :)

**Reply by dlxpbva** [reply]
Podobnie mam to rozwiązane teraz, dane leża na SP i mam tą biblioteke zsynchronizowaną na kompie. Minus taki, że muszę te dane fizycznie mieć na dysku aby je przeszukać co samo w sobie korzystając z PS'a jest czasochłonne (pojedyńcza CSV'ma ma około 150 MB).
Dlatego chciałem skorzystać z czegoś w pełni chmurowego, wtedy mogę do tego wystawić jakiś interfejs albo powerappkę i dać dostęp zainteresowanym.

**Comment by kaluzaaa** [comment]
- Idealne będzie to czego jeszcze nie ma... W Azure będzie taka usługa jak Azure Blob Quick Query, ale narazie publicznie cicho kiedy coś będzie.
- Azure Data Lake i Azure Synapse SQL (Preview) w trybie Serverless

**Reply by kaluzaaa** [reply]
a w sumie może zaraz pokażą Quick Query, bo jest już sample dla python -> https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-blob/samples/blob_samples_query.py

**Reply by wi5nia** [reply]
Mark mowil o tym na [sesji](https://medius.studios.ms/Embed/video-nc/B19-BRK3060?latestplayer=true&l=2077.222481) rok temu na Build :)

**Reply by dlxpbva** [reply]
Dzięki za sugestię, sprawdzę te rozwiązania.

**Comment by wi5nia** [comment]
Zawsze mozna isc troche ekstremalnie uzywajac Azure Searcha do [tego ](https://docs.microsoft.com/en-us/azure/search/search-howto-indexing-azure-blob-storage);)

**Comment by piotrgwiazda** [comment]
Możesz na jakiejś taniej vm-ce zainstalować Dremio w modelu wszystko na
jednej maszynie. Chwila zabawy jest z tym, ale potem robisz zapytania SQL

pon., 7 wrz 2020, 15:00 użytkownik Tomasz Wisniewski <
notifications@github.com> napisał:

**Comment by smereczynski** [comment]
Jak sporadycznie to Data Lake lub Blob i Databricks albo Synapse Serverless. Ewentualnie wspomniane przez @piotrgwiazda Dremio - bardzo dobra opcja. Można też bardziej egzotycznie z przerostem formy nad treścią, jak powoływanie klastra HDI na życzenie albo Azure Data Explorer 😄
